src/zoid/buttons/component.jsx

```diff
@@ -776,8 +801,29 @@ export const getButtonsComponent: () => ButtonsComponent = memoize(() => {
             messageType,
             offerCountryCode,
             creditProductIdentifier,
+            merchantID: serverMerchantId,
           }) => {
-            const { message, buttonSessionID, currency, merchantID } = props;
+            // merchantID that comes from props is an array
+            const {
+              message,
+              buttonSessionID,
+              currency,
+              merchantID: partnerMerchantId,
+            } = props;
+            const isPartnerMerchantIdEmpty =
+              !partnerMerchantId ||
+              partnerMerchantId.length === 0 ||
+              partnerMerchantId[0] === "";
+            // check to see if a partner merchant id integration exists
+            // if not grab the server merchant id
+            const merchantID = isPartnerMerchantIdEmpty
+              ? serverMerchantId
+              : partnerMerchantId?.join();
+
+            // override with server id if partner does not exist
+            getLogger().addTrackingBuilder(() => ({
+              [FPTI_KEY.SELLER_ID]: merchantID,
```

Comment:
Would something similar to `props.merchantID.toString() || serverMerchantId` be the same result without the need for some of the code above?

Reply:
yea that should work ill update